### PR TITLE
Support Joining a Game

### PR DESCRIPTION
**Pull request by rouxcaesar**
- Added support for joining an open game.
- Modified link on chess/index.html.erb page to trigger GamesController update method and redirect user to selected game.
- Added update action to GamesController that changes the open game's black_player to the id of the current user.
- Wrote an RSpec test for the update action, however it still fails for UrlGenerationError.

Help Request: If anyone can take a look at the my RSpec test and give advice on how to correct it for a true passing test, please let me know! Thanks.

### CODE REVIEW

**Review by dtrypt — COMMENTED** [review]
You didn't specify what game your trying to update in the test. You need something like id: game.id after the put :update. Also use :game instead of :fullgame because :fullgame is already populated with players. Here's what I think the test should look like:
  
```
describe 'games#update action' do
    it 'should successfully update the game black_player to the currently logged-in user id' do
      game = FactoryGirl.create(:game, black_player: nil)
      user = FactoryGirl.create(:user)
      put :update, id: game.id, game: { black_player: user.id }
      expect(response).to have_http_status(:success)
      game.reload
      expect(game.black_player).to eq(user.id)
    end
  end

```
It looks like there's some middleware issues with devise and rspec so you'll have to add a line to the rails_helper, see: http://stackoverflow.com/questions/38421853/why-is-my-rspec-not-loading-devisetestcontrollerhelpers

Lastly, I tried running the test and it's not successful. So I think that means that your update action in the games controller is not working properly so that will need to be fixed. I haven't had time to fully look into it...

UPDATE: It's failing because the user isn't signed in. This should work:

```
describe 'games#update action' do
    it 'should successfully update the game black_player to the currently logged-in user id' do
      game = FactoryGirl.create(:game, black_player: nil)
      user = FactoryGirl.create(:user)
      sign_in user
      put :update, id: game.id, game: { black_player: user.id }
      game.reload
      expect(game.black_player).to eq(user.id)
    end
  end

```
I've only reviewed your tests. Haven't looked at the rest of your code yet.

**Comment by sungwon** [comment]
Looks good so far! I'll take another look after the spec changes @trustinruss suggested.

**Comment by rouxcaesar** [comment]
I've made the changes suggested by Russ to the PR and it should be good for another review.

File: config/routes.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
   devise_for :users, controllers: { omniauth_callbacks: 'users/omniauth_callbacks' }
   root 'chess#index'
-  resources :games, only: [:new, :create, :show]
+  resources :games, only: [:new, :create, :show, :update]
   resources :pieces, only: [:show, :update]
+  get 'game', to: 'games#show'
```
Comment (sungwon) [inline config/routes.rb]: This seems to be redundant with line 4.

File: app/controllers/games_controller.rb [hunk 2]
```diff
@@ -14,9 +16,17 @@ def show
     render text: 'Not Found', status: :not_found if @game.blank?
   end
 
+  def update
+    @game = Game.find(params[:id])
+    if @game.black_player != current_user
```
Comment (sungwon) [inline app/controllers/games_controller.rb:21]: Is this just a sanity check to potentially avoid an unnecessary database call? cool.

**Review by sungwon — CHANGES_REQUESTED** [review]
I noticed that the #create method in the GamesController doesn't update the game by setting the white player as the current user. It's not part of the scope of your task, but if you'd like to take care of that while you're in the GamesController anyway, that'd be cool. I added a card for it in the backlog.

**Comment by sungwon** [comment]
Ah, I see you've already made a card for create game. Never mind about pulling it into this merge request then.

File: app/controllers/games_controller.rb (hunk 2 above)
Comment (rouxcaesar) [reply app/controllers/games_controller.rb:21]: Yes it is!

File: config/routes.rb (hunk 1 above)
Comment (rouxcaesar) [reply config/routes.rb]: Removed!

**Review by sungwon — APPROVED** [review]

**Review by dtrypt — APPROVED** [review]

**Pull request merged**